File: 2024/challenge-02/challenge-02.ts
function createFrame(names: string[]): string {
  const maxLength = Math.max(...names.map(name => name.length));
  const bordeSuperiorInferior = '*'.repeat(maxLength + 4);
  let marco = bordeSuperiorInferior + '\n';
  for (let nombre of names) {
    marco += `* ${nombre.padEnd(maxLength)} *\n`;
  }
  marco += bordeSuperiorInferior;
  return marco;
}

File: 2024/challenge-02/challenge-02.py
def createFrame(names):
    max_length = max(map(lambda x: len(x), names))
    borde_superior_inferior = '*' * (max_length + 4)
    marco = f'{borde_superior_inferior}\n'
    for nombre in names:
        marco += f'* {nombre.ljust(max_length)} *\n'
    marco += borde_superior_inferior
    return marco
